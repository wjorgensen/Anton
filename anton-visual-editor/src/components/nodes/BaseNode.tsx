'use client'

import { useState } from 'react'
import { Handle, Position, NodeProps } from '@xyflow/react'
import { AgentConfig, NodeStatus } from '@/types/agent'
import { Play, CheckCircle, XCircle, Loader2, Eye, Edit, Copy, Trash2, Settings } from 'lucide-react'
import { useFlowStore } from '@/store/flowStore'

interface BaseNodeData {
  agent: AgentConfig
  label: string
  instructions: string
  claudeMD: string
  status: NodeStatus
  inputs: Record<string, any>
  outputs: Record<string, any>
  config: {
    retryOnFailure: boolean
    maxRetries: number
    timeout: number
    requiresReview: boolean
  }
}

interface BaseNodeProps extends NodeProps<BaseNodeData> {
  categoryColor: string
}

const statusIcons: Record<NodeStatus, React.ReactNode> = {
  pending: <Play className="w-4 h-4" />,
  running: <Loader2 className="w-4 h-4 animate-spin" />,
  completed: <CheckCircle className="w-4 h-4" />,
  failed: <XCircle className="w-4 h-4" />,
  reviewing: <Eye className="w-4 h-4" />,
}

const statusColors: Record<NodeStatus, string> = {
  pending: 'text-text-secondary',
  running: 'text-accent-primary',
  completed: 'text-success',
  failed: 'text-error',
  reviewing: 'text-warning',
}

export default function BaseNode({ data, selected, categoryColor, id }: BaseNodeProps) {
  const { agent, label, status } = data
  const [isHovered, setIsHovered] = useState(false)
  const { selectNode, openEditModal, deleteNode, duplicateNode } = useFlowStore()
  
  const handleEdit = (e: React.MouseEvent) => {
    e.stopPropagation()
    openEditModal({ id, data, type: agent.category, position: { x: 0, y: 0 } })
  }
  
  const handleDuplicate = (e: React.MouseEvent) => {
    e.stopPropagation()
    duplicateNode(id)
  }
  
  const handleDelete = (e: React.MouseEvent) => {
    e.stopPropagation()
    deleteNode(id)
  }
  
  // Create gradient colors based on category
  const getGradient = (color: string) => {
    // Convert hex to RGB for gradient manipulation
    const hexToRgb = (hex: string) => {
      const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
      return result ? {
        r: parseInt(result[1], 16),
        g: parseInt(result[2], 16),
        b: parseInt(result[3], 16)
      } : null
    }
    
    const rgb = hexToRgb(color)
    if (!rgb) return color
    
    // Create a slightly darker version for gradient
    const darkerRgb = {
      r: Math.max(0, rgb.r - 40),
      g: Math.max(0, rgb.g - 40),
      b: Math.max(0, rgb.b - 40)
    }
    
    return `linear-gradient(135deg, ${color}, rgb(${darkerRgb.r}, ${darkerRgb.g}, ${darkerRgb.b}))`
  }
  
  return (
    <div
      className={`
        relative min-w-[200px] p-4 rounded-xl transition-all duration-300 shadow-lg group cursor-pointer
        ${selected ? 'ring-2 ring-offset-2 ring-offset-bg-primary transform scale-105 z-10' : ''}
        ${isHovered ? 'transform scale-102 z-20' : ''}
        ${status === 'running' ? 'animate-pulse' : ''}
        overflow-hidden backdrop-blur-sm
      `}
      style={{
        background: selected 
          ? getGradient(categoryColor)
          : isHovered 
            ? `linear-gradient(135deg, ${categoryColor}25, ${categoryColor}10)`
            : `linear-gradient(135deg, ${categoryColor}15, ${categoryColor}05)`,
        borderColor: selected ? categoryColor : isHovered ? `${categoryColor}50` : `${categoryColor}30`,
        ringColor: selected ? categoryColor : undefined,
        boxShadow: selected 
          ? `0 20px 40px ${categoryColor}40, 0 0 0 1px ${categoryColor}60`
          : isHovered
            ? `0 12px 40px rgba(0, 0, 0, 0.6), 0 0 0 1px ${categoryColor}50`
            : `0 8px 32px rgba(0, 0, 0, 0.4), 0 0 0 1px ${categoryColor}30`,
      }}
      onMouseEnter={() => setIsHovered(true)}
      onMouseLeave={() => setIsHovered(false)}
      onClick={() => selectNode(id)}
    >
      {/* Animated Background Pattern */}
      <div 
        className="absolute inset-0 opacity-5 pointer-events-none"
        style={{
          background: `repeating-linear-gradient(
            45deg,
            transparent,
            transparent 10px,
            ${categoryColor} 10px,
            ${categoryColor} 11px
          )`
        }}
      />
      
      {/* Status Glow Effect */}
      {status === 'running' && (
        <div 
          className="absolute inset-0 rounded-xl opacity-30 animate-pulse pointer-events-none"
          style={{
            background: `radial-gradient(circle at 50% 50%, ${categoryColor}60, transparent 70%)`,
          }}
        />
      )}
      
      {/* Quick Action Buttons */}
      <div className={`absolute -top-2 -right-2 flex gap-1 transition-all duration-200 ${
        isHovered || selected ? 'opacity-100 translate-y-0' : 'opacity-0 translate-y-2 pointer-events-none'
      }`}>
        <button
          onClick={handleEdit}
          className="w-8 h-8 bg-bg-secondary border border-border-primary rounded-full flex items-center justify-center hover:bg-bg-tertiary transition-colors shadow-lg backdrop-blur"
          title="Edit Node"
        >
          <Edit className="w-4 h-4" />
        </button>
        <button
          onClick={handleDuplicate}
          className="w-8 h-8 bg-bg-secondary border border-border-primary rounded-full flex items-center justify-center hover:bg-bg-tertiary transition-colors shadow-lg backdrop-blur"
          title="Duplicate Node"
        >
          <Copy className="w-4 h-4" />
        </button>
        <button
          onClick={handleDelete}
          className="w-8 h-8 bg-error/90 border border-error rounded-full flex items-center justify-center hover:bg-error transition-colors shadow-lg backdrop-blur text-white"
          title="Delete Node"
        >
          <Trash2 className="w-4 h-4" />
        </button>
      </div>
      
      {/* Enhanced Handles */}
      {agent.inputs.length > 0 && (
        <Handle
          type="target"
          position={Position.Left}
          className="w-5 h-5 transition-all duration-200 hover:scale-125 hover:shadow-glow"
          style={{ 
            background: selected ? '#ffffff' : categoryColor,
            border: `3px solid ${categoryColor}`,
            boxShadow: `0 0 15px ${categoryColor}60`,
            left: -10,
          }}
        />
      )}
      {agent.outputs.length > 0 && (
        <Handle
          type="source"
          position={Position.Right}
          className="w-5 h-5 transition-all duration-200 hover:scale-125 hover:shadow-glow"
          style={{ 
            background: selected ? '#ffffff' : categoryColor,
            border: `3px solid ${categoryColor}`,
            boxShadow: `0 0 15px ${categoryColor}60`,
            right: -10,
          }}
        />
      )}
      
      {/* Header */}
      <div className="relative z-10">
        <div className="flex items-center justify-between mb-3">
          <div className="flex items-center gap-3">
            <div className="flex items-center justify-center w-8 h-8 rounded-lg"
                 style={{ background: `${categoryColor}20` }}>
              <span className="text-xl">{agent.icon}</span>
            </div>
            <span className={`font-bold text-sm ${selected ? 'text-white' : 'text-text-primary'}`}>
              {label}
            </span>
          </div>
          <div className={`${statusColors[status]} transition-all duration-200`}>
            <div className="flex items-center justify-center w-6 h-6 rounded-full"
                 style={{ backgroundColor: status === 'running' ? `${categoryColor}20` : 'transparent' }}>
              {statusIcons[status]}
            </div>
          </div>
        </div>
        
        {/* Category Badge */}
        <div
          className="inline-flex items-center px-2 py-1 rounded-full text-xs font-semibold mb-3 backdrop-blur"
          style={{
            backgroundColor: selected ? 'rgba(255,255,255,0.2)' : `${categoryColor}25`,
            color: selected ? 'white' : categoryColor,
            border: `1px solid ${categoryColor}40`,
          }}
        >
          {agent.category.toUpperCase()}
        </div>
        
        {/* Description */}
        <p className={`text-xs leading-relaxed mb-3 ${selected ? 'text-white/90' : 'text-text-secondary'}`}>
          {agent.description}
        </p>
        
        {/* Status Bar */}
        <div className="flex items-center justify-between">
          {/* Review Badge */}
          {data.config.requiresReview && (
            <div className={`flex items-center gap-1.5 px-2 py-1 rounded-full text-xs ${
              selected ? 'text-white/80 bg-white/20' : 'text-warning bg-warning/10'
            }`}>
              <Eye className="w-3 h-3" />
              <span>Review Required</span>
            </div>
          )}
          
          {/* Connection Ports Info */}
          {(agent.inputs.length > 0 || agent.outputs.length > 0) && (
            <div className={`flex items-center gap-2 text-xs ${selected ? 'text-white/70' : 'text-text-muted'}`}>
              {agent.inputs.length > 0 && (
                <span className="flex items-center gap-1">
                  <div className="w-2 h-2 rounded-full" style={{ backgroundColor: categoryColor }}></div>
                  {agent.inputs.length} in
                </span>
              )}
              {agent.outputs.length > 0 && (
                <span className="flex items-center gap-1">
                  <div className="w-2 h-2 rounded-full" style={{ backgroundColor: categoryColor }}></div>
                  {agent.outputs.length} out
                </span>
              )}
            </div>
          )}
        </div>
      </div>
    </div>
  )
}