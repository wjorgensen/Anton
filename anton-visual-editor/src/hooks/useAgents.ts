import { useState, useEffect } from 'react'
import { AgentConfig, AgentCategory } from '@/types/agent'
import { loadAgents, getCachedAgents } from '@/lib/agents'

export function useAgents() {
  const [agents, setAgents] = useState<AgentConfig[]>(getCachedAgents())
  const [loading, setLoading] = useState(false)
  const [error, setError] = useState<string | null>(null)

  useEffect(() => {
    let mounted = true

    const loadAgentsAsync = async () => {
      setLoading(true)
      setError(null)
      
      try {
        const loadedAgents = await loadAgents()
        if (mounted) {
          setAgents(loadedAgents)
        }
      } catch (err) {
        if (mounted) {
          setError(err instanceof Error ? err.message : 'Failed to load agents')
        }
      } finally {
        if (mounted) {
          setLoading(false)
        }
      }
    }

    loadAgentsAsync()

    return () => {
      mounted = false
    }
  }, [])

  const getAgentsByCategory = (category: AgentCategory): AgentConfig[] => {
    return agents.filter(agent => agent.category === category)
  }

  const getAgentById = (id: string): AgentConfig | undefined => {
    return agents.find(agent => agent.id === id)
  }

  return {
    agents,
    loading,
    error,
    getAgentsByCategory,
    getAgentById,
  }
}