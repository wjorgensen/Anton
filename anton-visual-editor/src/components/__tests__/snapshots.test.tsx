import React from 'react'
import { render } from '@testing-library/react'
import FlowEditor from '../FlowEditor'
import AgentLibrary from '../AgentLibrary'
import NodeEditModal from '../NodeEditModal'
import PreviewManager from '../PreviewManager'
import BaseNode from '../nodes/BaseNode'
import ReviewNode from '../nodes/ReviewNode'
import { useFlowStore } from '@/store/flowStore'
import { useAgents } from '@/hooks/useAgents'
import { useWebSocket } from '@/hooks/useWebSocket'

// Mock stores and hooks
jest.mock('@/store/flowStore')
jest.mock('@/hooks/useAgents')
jest.mock('@/hooks/useWebSocket')

describe('Component Snapshots', () => {
  const mockUseFlowStore = useFlowStore as jest.MockedFunction<typeof useFlowStore>
  const mockUseAgents = useAgents as jest.MockedFunction<typeof useAgents>
  const mockUseWebSocket = useWebSocket as jest.MockedFunction<typeof useWebSocket>
  
  beforeEach(() => {
    jest.clearAllMocks()
    
    // Setup default mocks
    mockUseFlowStore.mockReturnValue({
      nodes: [
        {
          id: 'node-1',
          type: 'execution',
          position: { x: 100, y: 100 },
          data: {
            label: 'Test Node',
            agentConfig: {
              id: 'test-agent',
              name: 'Test Agent',
              category: 'execution',
              description: 'Test description',
            },
          },
        },
      ],
      edges: [],
      editingNode: null,
      selectedNode: null,
      onNodesChange: jest.fn(),
      onEdgesChange: jest.fn(),
      onConnect: jest.fn(),
      addNode: jest.fn(),
      deleteNode: jest.fn(),
      selectNode: jest.fn(),
      openEditModal: jest.fn(),
      closeEditModal: jest.fn(),
      updateNode: jest.fn(),
      duplicateNode: jest.fn(),
      clearFlow: jest.fn(),
    })
    
    mockUseAgents.mockReturnValue({
      agents: [
        {
          id: 'agent-1',
          name: 'Sample Agent',
          category: 'setup',
          type: 'setup',
          version: '1.0.0',
          description: 'Sample agent for testing',
          icon: 'ü§ñ',
          color: '#3B82F6',
          instructions: { base: 'Test', contextual: 'Test' },
          inputs: [],
          outputs: [],
          tags: ['test'],
        },
      ],
      loading: false,
      error: null,
      getAgentsByCategory: jest.fn(),
      getAgentById: jest.fn(),
    })
    
    mockUseWebSocket.mockReturnValue({
      connected: true,
      connecting: false,
      error: null,
      nodeUpdates: {},
      previewData: {
        terminal: {
          nodeId: 'node-1',
          output: 'Sample terminal output',
          timestamp: new Date().toISOString(),
        },
      },
      subscribeToExecution: jest.fn(),
      subscribeToProject: jest.fn(),
      reconnect: jest.fn(),
    })
  })

  describe('FlowEditor Snapshots', () => {
    it('matches snapshot with empty flow', () => {
      mockUseFlowStore.mockReturnValue({
        ...mockUseFlowStore(),
        nodes: [],
        edges: [],
      })
      
      const { container } = render(<FlowEditor />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('matches snapshot with nodes and edges', () => {
      mockUseFlowStore.mockReturnValue({
        ...mockUseFlowStore(),
        nodes: [
          {
            id: 'node-1',
            type: 'execution',
            position: { x: 100, y: 100 },
            data: { label: 'Node 1', agentConfig: { name: 'Agent 1' } },
          },
          {
            id: 'node-2',
            type: 'review',
            position: { x: 300, y: 100 },
            data: { label: 'Node 2', agentConfig: { name: 'Agent 2' } },
          },
        ],
        edges: [
          { id: 'edge-1', source: 'node-1', target: 'node-2' },
        ],
      })
      
      const { container } = render(<FlowEditor />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('matches snapshot with selected node', () => {
      mockUseFlowStore.mockReturnValue({
        ...mockUseFlowStore(),
        selectedNode: 'node-1',
      })
      
      const { container } = render(<FlowEditor />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('AgentLibrary Snapshots', () => {
    it('matches snapshot with agents', () => {
      const mockOnDragStart = jest.fn()
      const { container } = render(<AgentLibrary onDragStart={mockOnDragStart} />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('matches snapshot while loading', () => {
      mockUseAgents.mockReturnValue({
        ...mockUseAgents(),
        agents: [],
        loading: true,
      })
      
      const mockOnDragStart = jest.fn()
      const { container } = render(<AgentLibrary onDragStart={mockOnDragStart} />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('matches snapshot with error state', () => {
      mockUseAgents.mockReturnValue({
        ...mockUseAgents(),
        agents: [],
        error: 'Failed to load agents',
      })
      
      const mockOnDragStart = jest.fn()
      const { container } = render(<AgentLibrary onDragStart={mockOnDragStart} />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('Node Component Snapshots', () => {
    it('BaseNode matches snapshot', () => {
      const props = {
        id: 'base-1',
        data: {
          label: 'Base Node',
          agentConfig: {
            name: 'Base Agent',
            description: 'Base agent description',
            category: 'setup',
            icon: '‚öôÔ∏è',
          },
        },
        selected: false,
        categoryColor: '#3B82F6',
      }
      
      const { container } = render(<BaseNode {...props} />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('BaseNode matches snapshot when selected', () => {
      const props = {
        id: 'base-1',
        data: {
          label: 'Base Node',
          agentConfig: {
            name: 'Base Agent',
            description: 'Base agent description',
            category: 'setup',
            icon: '‚öôÔ∏è',
          },
        },
        selected: true,
        categoryColor: '#3B82F6',
      }
      
      const { container } = render(<BaseNode {...props} />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('ReviewNode matches snapshot in pending state', () => {
      const props = {
        id: 'review-1',
        data: {
          label: 'Review Node',
          agentConfig: {
            name: 'Review Agent',
            description: 'Review agent description',
          },
          status: 'pending' as const,
          reviewScope: 'full' as const,
          requiresApproval: true,
        },
        selected: false,
      }
      
      const { container } = render(<ReviewNode {...props} />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('ReviewNode matches snapshot in approved state', () => {
      const props = {
        id: 'review-1',
        data: {
          label: 'Review Node',
          agentConfig: {
            name: 'Review Agent',
            description: 'Review agent description',
          },
          status: 'approved' as const,
          reviewScope: 'changes' as const,
          feedback: [
            {
              id: 'feedback-1',
              message: 'Looks good!',
              severity: 'info' as const,
              timestamp: new Date().toISOString(),
            },
          ],
        },
        selected: false,
      }
      
      const { container } = render(<ReviewNode {...props} />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('NodeEditModal Snapshots', () => {
    it('matches snapshot when open', () => {
      mockUseFlowStore.mockReturnValue({
        ...mockUseFlowStore(),
        editingNode: {
          id: 'node-1',
          type: 'execution',
          position: { x: 100, y: 100 },
          data: {
            label: 'Edit Node',
            agentConfig: {
              id: 'agent-1',
              name: 'Edit Agent',
              description: 'Agent being edited',
              inputs: [
                { name: 'input1', type: 'string', required: true },
                { name: 'input2', type: 'number', required: false },
              ],
              outputs: [
                { name: 'output1', type: 'string' },
              ],
            },
            inputs: { input1: 'test' },
            timeout: 30000,
          },
        },
      })
      
      const { container } = render(<NodeEditModal />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('matches snapshot when closed', () => {
      mockUseFlowStore.mockReturnValue({
        ...mockUseFlowStore(),
        editingNode: null,
      })
      
      const { container } = render(<NodeEditModal />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('PreviewManager Snapshots', () => {
    it('matches snapshot with terminal output', () => {
      const { container } = render(<PreviewManager executionId="exec-1" />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('matches snapshot with web preview', () => {
      mockUseWebSocket.mockReturnValue({
        ...mockUseWebSocket(),
        previewData: {
          web: {
            nodeId: 'node-1',
            url: 'http://localhost:3000',
            html: '<html><body>Preview</body></html>',
            timestamp: new Date().toISOString(),
          },
        },
      })
      
      const { container } = render(<PreviewManager executionId="exec-1" />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('matches snapshot when disconnected', () => {
      mockUseWebSocket.mockReturnValue({
        ...mockUseWebSocket(),
        connected: false,
        error: 'Connection failed',
      })
      
      const { container } = render(<PreviewManager executionId="exec-1" />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('Dark Theme Snapshots', () => {
    beforeEach(() => {
      document.documentElement.classList.add('dark')
    })

    afterEach(() => {
      document.documentElement.classList.remove('dark')
    })

    it('FlowEditor matches dark theme snapshot', () => {
      const { container } = render(<FlowEditor />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('AgentLibrary matches dark theme snapshot', () => {
      const mockOnDragStart = jest.fn()
      const { container } = render(<AgentLibrary onDragStart={mockOnDragStart} />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('BaseNode matches dark theme snapshot', () => {
      const props = {
        id: 'base-1',
        data: {
          label: 'Base Node',
          agentConfig: {
            name: 'Base Agent',
            description: 'Base agent description',
            category: 'setup',
            icon: '‚öôÔ∏è',
          },
        },
        selected: false,
        categoryColor: '#3B82F6',
      }
      
      const { container } = render(<BaseNode {...props} />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})