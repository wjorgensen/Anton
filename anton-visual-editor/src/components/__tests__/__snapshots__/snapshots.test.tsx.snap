// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Component Snapshots AgentLibrary Snapshots matches snapshot while loading 1`] = `
<div
  class="w-80 bg-bg-secondary border-r border-border p-4 space-y-4"
>
  <div
    class="space-y-2"
  >
    <div
      class="bg-border rounded animate-pulse h-8 w-32"
    />
    <div
      class="bg-border rounded animate-pulse h-10 w-full rounded"
    />
  </div>
  <div
    class="space-y-2"
  >
    <div
      class="flex items-center gap-2"
    >
      <div
        class="bg-border rounded animate-pulse w-4 h-4"
      />
      <div
        class="bg-border rounded animate-pulse h-5 w-24"
      />
      <div
        class="bg-border rounded animate-pulse w-6 h-4 rounded-full ml-auto"
      />
    </div>
    <div
      class="pl-6 space-y-2"
    >
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="space-y-2"
  >
    <div
      class="flex items-center gap-2"
    >
      <div
        class="bg-border rounded animate-pulse w-4 h-4"
      />
      <div
        class="bg-border rounded animate-pulse h-5 w-24"
      />
      <div
        class="bg-border rounded animate-pulse w-6 h-4 rounded-full ml-auto"
      />
    </div>
    <div
      class="pl-6 space-y-2"
    >
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="space-y-2"
  >
    <div
      class="flex items-center gap-2"
    >
      <div
        class="bg-border rounded animate-pulse w-4 h-4"
      />
      <div
        class="bg-border rounded animate-pulse h-5 w-24"
      />
      <div
        class="bg-border rounded animate-pulse w-6 h-4 rounded-full ml-auto"
      />
    </div>
    <div
      class="pl-6 space-y-2"
    >
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-2"
      >
        <div
          class="bg-border rounded animate-pulse w-8 h-8 rounded"
        />
        <div
          class="flex-1 space-y-1"
        >
          <div
            class="bg-border rounded animate-pulse h-3 w-32"
          />
          <div
            class="bg-border rounded animate-pulse h-2 w-24"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component Snapshots AgentLibrary Snapshots matches snapshot with error state 1`] = `
<div
  class="fixed inset-0 bg-black/50 z-40 md:hidden transition-opacity duration-300 opacity-0 pointer-events-none"
/>
`;

exports[`Component Snapshots Node Component Snapshots ReviewNode matches snapshot in approved state 1`] = `
<div
  class="
        relative min-w-[250px] rounded-lg border-2 transition-all duration-200
        border-border-primary
        bg-bg-secondary hover:bg-bg-tertiary
      "
>
  <div
    class="!bg-accent-primary !w-3 !h-3 !border-2 !border-bg-primary"
    data-testid="react-flow-handle"
    position="left"
    type="target"
  />
  <div
    class="px-3 py-2 border-b border-border-primary rounded-t-md"
    style="background-color: rgba(16, 185, 129, 0.125);"
  >
    <div
      class="flex items-center justify-between"
    >
      <div
        class="flex items-center gap-2"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-circle-check w-4 h-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(16, 185, 129);"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="m9 12 2 2 4-4"
          />
        </svg>
        <span
          class="text-xs font-medium text-text-secondary"
        >
          Approved
        </span>
      </div>
    </div>
  </div>
  <div
    class="px-3 py-2"
  >
    <h3
      class="text-sm font-semibold text-text-primary mb-1"
    >
      Review Node
    </h3>
    <div
      class="text-xs text-text-secondary mb-2"
    >
      Scope: 
      <span
        class="text-accent-primary capitalize"
      >
        changes
      </span>
    </div>
    <div
      class="flex items-center gap-3 mt-2 pt-2 border-t border-border-primary"
    >
      <div
        class="flex items-center gap-1"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-message-square w-3 h-3 text-info"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 17a2 2 0 0 1-2 2H6.828a2 2 0 0 0-1.414.586l-2.202 2.202A.71.71 0 0 1 2 21.286V5a2 2 0 0 1 2-2h16a2 2 0 0 1 2 2z"
          />
        </svg>
        <span
          class="text-xs text-text-secondary"
        >
          1
           comments
        </span>
      </div>
    </div>
  </div>
  <div
    class="!bg-accent-primary !w-3 !h-3 !border-2 !border-bg-primary"
    data-testid="react-flow-handle"
    position="right"
    type="source"
  />
</div>
`;

exports[`Component Snapshots Node Component Snapshots ReviewNode matches snapshot in pending state 1`] = `
<div
  class="
        relative min-w-[250px] rounded-lg border-2 transition-all duration-200
        border-border-primary
        bg-bg-secondary hover:bg-bg-tertiary
      "
>
  <div
    class="!bg-accent-primary !w-3 !h-3 !border-2 !border-bg-primary"
    data-testid="react-flow-handle"
    position="left"
    type="target"
  />
  <div
    class="px-3 py-2 border-b border-border-primary rounded-t-md"
    style="background-color: rgba(102, 102, 102, 0.125);"
  >
    <div
      class="flex items-center justify-between"
    >
      <div
        class="flex items-center gap-2"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-clock w-4 h-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(102, 102, 102);"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 6v6l4 2"
          />
          <circle
            cx="12"
            cy="12"
            r="10"
          />
        </svg>
        <span
          class="text-xs font-medium text-text-secondary"
        >
          Pending Review
        </span>
      </div>
      <svg
        class="lucide lucide-circle-alert w-4 h-4 text-warning"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        title="Requires manual approval"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="8"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="16"
          y2="16"
        />
      </svg>
    </div>
  </div>
  <div
    class="px-3 py-2"
  >
    <h3
      class="text-sm font-semibold text-text-primary mb-1"
    >
      Review Node
    </h3>
    <div
      class="text-xs text-text-secondary mb-2"
    >
      Scope: 
      <span
        class="text-accent-primary capitalize"
      >
        full
      </span>
    </div>
  </div>
  <div
    class="!bg-accent-primary !w-3 !h-3 !border-2 !border-bg-primary"
    data-testid="react-flow-handle"
    position="right"
    type="source"
  />
</div>
`;

exports[`Component Snapshots NodeEditModal Snapshots matches snapshot when closed 1`] = `null`;

exports[`Component Snapshots NodeEditModal Snapshots matches snapshot when open 1`] = `null`;
